homeassistant/components/discovergy/sensor.py

```diff
@@ -0,0 +1,145 @@
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            return await discovergy_instance.get_last_reading(meter.get_meter_id())
+        except AccessTokenExpired as err:
+            raise ConfigEntryAuthFailed(
+                "Got token expired while communicating with API"
+            ) from err
+        except HTTPError as err:
+            raise UpdateFailed(f"Error communicating with API: {err}") from err
+        except Exception as err:  # pylint: disable=broad-except
+            raise UpdateFailed(
+                f"Unexpected error while communicating with API: {err}"
+            ) from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="sensor",
+        update_method=async_update_data,
+        update_interval=MIN_TIME_BETWEEN_UPDATES,
+    )
+    return coordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Discovergy sensors."""
+    try:
+        discovergy_instance = hass.data[DOMAIN][entry.entry_id]
+        meters = await discovergy_instance.get_meters()
+    except AccessTokenExpired as err:
+        _LOGGER.debug("Token expired while communicating with API: %s", err)
+        entry.async_start_reauth(hass)
+    except HTTPError as err:
+        raise ConfigEntryNotReady(f"Error communicating with API: {err}") from err
+    except Exception as err:  # pylint: disable=broad-except
+        raise ConfigEntryNotReady(
+            f"Unexpected error while communicating with API: {err}"
+        ) from err
+    else:
+        entities = []
+        for meter in meters:
+            if meter.measurement_type == "ELECTRICITY":
+                # Get coordinator for meter, set config entry and fetch initial data
+                # so we have data when entities are added
+                coordinator = get_coordinator_for_meter(
+                    hass, meter, discovergy_instance
+                )
+                coordinator.config_entry = entry
+                await coordinator.async_config_entry_first_refresh()
+
+                for description in ELECTRICITY_SENSORS:
+                    # check if this meter has this data, then add this sensor
+                    if coordinator.data.values[description.key]:
+                        entities.append(
+                            DiscovergyElectricitySensor(description, meter, coordinator)
+                        )
+
+        async_add_entities(entities, False)
+
+
+class DiscovergyElectricitySensor(CoordinatorEntity, SensorEntity):
+    """Represents a discovergy electricity smart meter sensor."""
+
+    def __init__(
+        self,
+        description: SensorEntityDescription,
+        meter: Meter,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+
+        self._meter = meter
```

Comment:
```suggestion
```

Reply:
Indeed, this is not needed anymore. My bad.